Can't compile on Linux with glibc 2.32 (sys_siglist deprecated)

Quoting from the glibc 2.32 [changelog](https://sourceware.org/pipermail/libc-announce/2020/000029.html):

    The deprecated arrays sys_siglist, _sys_siglist, and sys_sigabbrev
    are no longer available to newly linked binaries, and their declarations
    have been removed from <string.h>.  They are exported solely as
    compatibility symbols to support old binaries.  All programs should use
    strsignal instead.

I assume because the symbol still exists, `check_symbol_exists(sys_siglist "signal.h" HAVE_DECL_SYS_SIGLIST)` in CMakeLists.txt is returning true, resulting in a compilation error.

    src/prboom-plus/prboom2/src/POSIX/i_system.c:129:14: error: ‘sys_siglist’ undeclared (first use in this function)
      129 |   if (strlen(sys_siglist[signum]) < sz)

Temp workaround for anyone trying to compile is to change `#ifdef HAVE_DECL_SYS_SIGLIST` in src/POSIX/i_system.c and src/SDL/i_system.c to `#if 0`

--- a/prboom2/CMakeLists.txt
+++ b/prboom2/CMakeLists.txt
@@ -48,7 +48,7 @@ if(NOT WIN32)
     endif()
 endif()
 check_symbol_exists(usleep "unistd.h" HAVE_USLEEP)
-check_symbol_exists(sys_siglist "signal.h" HAVE_DECL_SYS_SIGLIST)
+check_symbol_exists(strsignal "string.h" HAVE_STRSIGNAL)
 
 include(CheckIncludeFile)
 

--- a/prboom2/cmake/config.h.cin
+++ b/prboom2/cmake/config.h.cin
@@ -11,7 +11,7 @@
 #cmakedefine HAVE_MMAP
 #cmakedefine HAVE_SCHED_SETAFFINITY
 #cmakedefine HAVE_USLEEP
-#cmakedefine HAVE_DECL_SYS_SIGLIST
+#cmakedefine HAVE_STRSIGNAL
 
 #cmakedefine HAVE_SYS_WAIT_H
 #cmakedefine HAVE_UNISTD_H

--- a/prboom2/src/POSIX/i_system.c
+++ b/prboom2/src/POSIX/i_system.c
@@ -45,6 +45,7 @@
 #include <sys/time.h>
 #include <ctype.h>
 #include <signal.h>
+#include <string.h>
 
 #include "doomtype.h"
 #include "m_fixed.h"
@@ -125,9 +126,9 @@ const char* I_GetVersionString(char* buf, size_t sz)
  */
 const char* I_SigString(char* buf, size_t sz, int signum)
 {
-#ifdef HAVE_DECL_SYS_SIGLIST
-  if (strlen(sys_siglist[signum]) < sz)
-    strcpy(buf,sys_siglist[signum]);
+#ifdef HAVE_STRSIGNAL
+  if (strsignal(signum) && strlen(strsignal(signum)) < sz)
+    strcpy(buf,strsignal(signum));
   else
 #endif
     sprintf(buf,"signal %d",signum);

--- a/prboom2/src/SDL/i_system.c
+++ b/prboom2/src/SDL/i_system.c
@@ -45,6 +45,7 @@
 #include <ctype.h>
 #include <time.h>
 #include <signal.h>
+#include <string.h>
 #ifdef _MSC_VER
 #define    F_OK    0    /* Check for file existence */
 #define    W_OK    2    /* Check for write permission */
@@ -219,9 +220,9 @@ const char* I_GetVersionString(char* buf, size_t sz)
  */
 const char* I_SigString(char* buf, size_t sz, int signum)
 {
-#ifdef HAVE_DECL_SYS_SIGLIST
-  if (strlen(sys_siglist[signum]) < sz)
-    strcpy(buf,sys_siglist[signum]);
+#ifdef HAVE_STRSIGNAL
+  if (strsignal(signum) && strlen(strsignal(signum)) < sz)
+    strcpy(buf,strsignal(signum));
   else
 #endif
   snprintf(buf,sz,"signal %d",signum);

